src/mcmc/abstractmcmc.jl

```diff
@@ -1,15 +1,352 @@
-# TODO: Implement additional checks for certain samplers, e.g.
-# HMC not supporting discrete parameters.
-function _check_model(model::DynamicPPL.Model)
-    return DynamicPPL.check_model(model; error_on_failure=true)
+# This file contains the basic methods for `AbstractMCMC.sample`.
+# The overall aim is that users can call
+#
+#    sample(::Model, ::InferenceAlgorithm, N)
```

Comment:
I assume `InferenceAlgorithm` here is still needed until we update all the interface?